src/legacy/core_plugins/tagcloud/public/tag_cloud_fn.ts

```diff
@@ -63,25 +60,26 @@ export const tagcloud = () => ({
     metric: {
       types: ['vis_dimension'],
       help: i18n.translate('tagCloud.function.metric.help', {
-        defaultMessage: 'metric dimension configuration'
+        defaultMessage: 'metric dimension configuration',
       }),
       required: true,
     },
     bucket: {
       types: ['vis_dimension'],
       help: i18n.translate('tagCloud.function.bucket.help', {
-        defaultMessage: 'bucket dimension configuration'
+        defaultMessage: 'bucket dimension configuration',
       }),
     },
   },
-  fn(context, args) {
+  fn(context: any, args: any) {
```

Comment:
If we are converting this function to TS, we should use the `ExpressionFunction` generic from `src/legacy/core_plugins/interpreter/public`, e.g. 
```ts
export const createTagCloudFn = (): ExpressionFunction<'tagcloud', Context, Arguments, Return> => ({
```

This will automatically assign the correct values to `context` and `args` for you. 

We are going to have to do this anyway, but I'm also fine with converting these to TS in a subsequent PR; it's just if we are going to convert them, I prefer to do it thoroughly so we don't need to revisit.

For the parameters on this function, the `Arguments` interface is something you can define that would be internal to this file. `Context` will be what is specified on line 26, which you can actually import from the interpreter. And `Return` is what is specified by `type: 'render'` on line 24.

So altogether would look something like this:
```ts
import { ExpressionFunction, KibanaDatatable, Render } from '../../interpreter/types';

const name = 'tagcloud';

type Context = KibanaDatatable;

interface Arguments {
  scale: string;
  orientation: string;
  minFontSize: number;
  maxFontSize: number;
  showLabel: boolean;
  metric: any; // these aren't typed yet
  bucket: any; // these aren't typed yet
}

type VisParams = Omit<Arguments, 'bucket'>;

interface RenderValue {
  visType: typeof name;
  visData: Context;
  visConfig: VisParams;
  params: any;
}

type Return = Promise<Render<RenderValue>>;

export const createTagCloudFn = (): ExpressionFunction<typeof name, Context, Arguments, Return> => ({
...
```